paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -169,13 +174,30 @@ public class PaparazziPlugin @Inject constructor(
       val isRecordRun = project.objects.property(Boolean::class.java)
       val isVerifyRun = project.objects.property(Boolean::class.java)
 
+      val snapshotFailures = failedSnapshotOutputDir.flatMap {
+        project.objects.directoryProperty().apply {
+          set(if (it.asFile.exists()) it else null)
+        }
+      }
+
       project.gradle.taskGraph.whenReady { graph ->
         isRecordRun.set(recordTaskProvider.map { graph.hasTask(it) })
         isVerifyRun.set(verifyTaskProvider.map { graph.hasTask(it) })
+
+        // Override test reporter with custom one to include diff screenshots in the report.
+        if (isVerifyRun.get()) {
```

Comment:
reminder: we def don't want to add anything else in this callback.

Let's use the same lambda approach as we did in the Gradle caching PR?  If that doesn't work, let's spike some design strategies in this thread!